cmd/jaeger/internal/processors/adaptivesampling/processor.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package adaptivesampling
+
+import (
+	"context"
+	"fmt"
+
+	otlp2jaeger "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/extension/remotesampling"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"github.com/jaegertracing/jaeger/plugin/sampling/strategystore/adaptive"
+)
+
+type traceProcessor struct {
+	config     *Config
+	logger     *zap.Logger
+	aggregator strategystore.Aggregator
+}
+
+func newTraceProcessor(cfg Config, otel component.TelemetrySettings) *traceProcessor {
+	return &traceProcessor{
+		config: &cfg,
+		logger: otel.Logger,
+	}
+}
+
+func (tp *traceProcessor) start(_ context.Context, host component.Host) error {
+	extCfg, err := remotesampling.GetExtensionConfig(host)
```

Comment:
Sorry for bit delay

> Can you create a README with a diagram showing which components are running inside processor and extension? 


yes


> is aggregator needed for the extension?


no

> why do you need to call extension to create the aggregator?


i am calling `GetExtensionConfig`, just to get these configs (to avoid config duplication) : `InitialSamplingProbability`, `AggregationBuckets`, `LeaderLeaseRefreshInterval`, `FollowerLeaseRefreshInterval`.

> I think extension only need to instantiate the storage.

Yes, it instantiates a sampling store, loads probabilities from it, and then serves responses through HTTP and gRPC.


I think the confusion may arise due to the naming of this 'GetExtensionConfig'. Its only purpose is to share `InitialSamplingProbability`, `AggregationBuckets`, `LeaderLeaseRefreshInterval`, and `FollowerLeaseRefreshInterval'` with the processor.

Reply:
> Its only purpose is to share InitialSamplingProbability, AggregationBuckets, LeaderLeaseRefreshInterval, and FollowerLeaseRefreshInterval' with the processor.

Does the processor actually need these values directly or does it need an object that is already built with these settings like the leader/follower manager?

Reply:
[Needed Directly] InitialSamplingProbability: // used in postAggregator.calculateProbability()
		AggregationBuckets: // used in sampling store initialization
		LeaderLeaseRefreshInterval: // used in leader election
		FollowerLeaseRefreshInterval: // used in leader election

Reply:
see https://github.com/jaegertracing/jaeger/pull/5389#discussion_r1623307763

That comment takes care of 3 out of 4 flags, leaving only `InitialSamplingProbability`. That one I do not see being used by the query-side of adaptive sampling , only by the processor, so we can move the parameter there
```
plugin/sampling//strategystore/adaptive/processor.go:398:	oldProbability := p.InitialSamplingProbability
plugin/sampling//strategystore/adaptive/processor.go:450:	if FloatEquals(probability, p.InitialSamplingProbability) {
plugin/sampling//strategystore/adaptive/processor.go:466:			return e.UsingAdaptive && !FloatEquals(e.Probability, p.InitialSamplingProbability)
```